### PR TITLE
It is safer to pylint on Python 3 than Python 2

### PR DESCRIPTION
**Pull request by cclauss**
Syntax Errors will be caught by pylint running on Python 3 but not when running on Python 2:
* print "Hello"
* 0L
* 0755
* except Exception, e
etc...

### CODE REVIEW

**Comment by cclauss** [comment]
### The really strange case of __python3 -m pylint import_os.py__...

$ __git clone https://github.com/certbot/certbot__
$ __cd certbot__
$ __cat > import_os.py__
$ __python3 -m pylint import_os.py__
> import os
> \<Ctrl-D>

$ __python3 -m pylint import_os.py__
> internal error with sending report for module ['import_os.py']
> The concept of slots is undefined for old-style classes.

$ __cp import_os.py ..__
$ __cd ..__
$ __python3 -m pylint import_os.py__
> Pylint works as expected

Conclusion: There is not a problem with your Python files, there is a problem with your error reporting system.  If you try the same trick with __importt_sys.py__ things work so there is really something about __os__ like perhaps there is an __os.py__ file lurking around somewhere or something.

**Comment by cclauss** [comment]
> we rely on clean lint results for new code

Not really.  Pylint dirt is currently being swept under the rug.

This PR demonstrates that for a while now Pylint has not been running properly either locally or on Travis because of __linter_plugin.py__.  Once that is fixed, the Pylint errors are properly reported.

The PR has now reverted to running Pylint on Python 2 and the repo currently contains 90+ Python files that generate Pylint errors.  I am not inclined to fix all those files but it might make a nice set of tasks for first time contributors.

When __linter_plugin.py__ is deleted then the __apache_compat__ and __nginx_compat__ test runs fail.  So __linter_plugin.py__ is fixed to be valid Python and the __cls.slots()__ bit commented out.  __.pylintrc__ is modified to no longer load the plugin.

File: tox.ini [hunk 1]
```diff
@@ -133,6 +133,7 @@ basepython = python2.7
 # continue, but tox return code will reflect previous error
 commands =
     {[base]install_packages}
+    pip install --upgrade astroid pylint  # required!
```
[the remark this answers is not in the document]
Comment (cclauss) [reply tox.ini]: Try taking it out and see what breaks.

[the remark this answers is not in the document]
Comment (ohemorange) [reply tox.ini]: `tools/dev_constraints.txt` should also be updated

**Comment by bmw** [comment]
@ohemorange, are you happy with this PR besides fixing the new linting failures?

**Comment by ohemorange** [comment]
@bmw no, there's still an unaddressed review comment that I'd like to see cleared up.

**Comment by bmw** [comment]
OK. The fix here is to remove the line commented on at https://github.com/certbot/certbot/pull/5727#discussion_r184552864 and update the pinned versions in `tools/dev_constraints.txt`.

@sydneyli, perhaps @jepayne1138 can do this in #6008?

**Comment by ohemorange** [comment]
Closing this in favor of #6008. @cclauss if you can make the change described at https://github.com/certbot/certbot/pull/5727#discussion_r184552864, then we'll reopen to finish it up here. Thanks for all the great work you've done so far!

**Pull request closed without merging**